fe/fe-core/src/main/java/com/starrocks/common/ConfigBase.java

```diff
@@ -75,29 +79,34 @@ public class ConfigBase {
         String[] aliases() default {};
     }
 
-    protected Properties props;
+    protected Properties props = new Properties();
+    private static String mutableConfigPath;
     protected static Field[] configFields;
     protected static Map<String, Field> allMutableConfigs = new HashMap<>();
 
     public void init(String propFile) throws Exception {
         configFields = this.getClass().getFields();
-        initAllMutableConfigs();
-        props = new Properties();
-        FileReader reader = null;
-        try {
-            reader = new FileReader(propFile);
+        try (FileReader reader = new FileReader(propFile)) {
             props.load(reader);
-        } catch (Exception e) {
-            throw e;
-        } finally {
-            if (reader != null) {
-                reader.close();
-            }
         }
         replacedByEnv();
+    }
+
+    public void initMutable(String propFile) throws Exception {
```

Comment:
what if the conf/ directory is not writable?

Reply:
By default, files are created with read and write permissions. If a user modifies the permissions, a permission exception will be thrown during persistence.

Reply:
Should not take this as assumption. Think about if the service is running in a container with strict security constraints. How to design/implement this feature in the specific scenario? or the FE service can know that it is not writable, and the feature will be disabled by default instead of throw exception to end user.